src/Workspaces/Remote/Core/BrokeredServiceConnection.cs

```diff
@@ -138,8 +138,8 @@ public override async ValueTask<Optional<TResult>> TryInvokeAsync<TResult>(Solut
             var (clientStream, serverStream) = FullDuplexStream.CreatePair();
 
             // Create new tasks that both start executing, rather than invoking the delegates directly.
-            // If the reader started synchronously reading before the writer task started it would be blocking, and vice versa
-            // if the writer synchronously filled the buffer before the reader task started it would also be blocking.
+            // If the reader started synchronously reading before the writer task started it would hang, and vice versa
+            // if the writer synchronously filled the buffer before the reader task started it would also hang.
```

Comment:
```suggestion
            // If the reader started synchronously reading before the writer task started it would deadlock, and vice versa
            // if the writer synchronously filled the buffer before the reader task started it would also deadlock.
```